docs/contributing/add_new_resource_tgc_next.md

```diff
@@ -0,0 +1,330 @@
+
+### Getting a terraform resource name from a GCP resource name
+
+The first step in determining if a GCP resource is supported is to figure out the name of the corresponding Terraform resource. You can often do this by searching for the GCP resource name in the [Terraform google-beta provider documentation](https://registry.terraform.io/providers/hashicorp/google-beta/latest/docs).
+
+### Getting the CAI asset type of a GCP resource
+
+The second step in determining if a GCP resource is supported is to verify its inclusion in the Cloud Asset Inventory (CAI). You can complete this by searching for the resource's asset type within CAI's list of [supported asset types](https://cloud.google.com/asset-inventory/docs/asset-types).
+
+## How to add support for a new resource
+
+To fully support a new resource in the terraform-google-conversion tool, you must implement support for it in both tfplan2cai and cai2hcl.
+
+`tfplan2cai` Support: The resource must have an entry in the [`pkg/tfplan2cai/converters/resource_converters.go` file](https://github.com/GoogleCloudPlatform/terraform-google-conversion/blob/main/pkg/tfplan2cai/converters/resource_converters.go). You can confirm this by searching that file for the Terraform resource name, such as `google_alloydb_backup`.
+
+`cai2hcl` Support: The resource must have an entry in the [`pkg/cai2hcl/converters/resource_converters.go` file](https://github.com/GoogleCloudPlatform/terraform-google-conversion/blob/main/pkg/cai2hcl/converters/resource_converters.go). You can verify this by searching for the corresponding Cloud Asset type, such as `alloydb.googleapis.com/Backup`.
+
+Adding support for a resource has 4 steps:
+
+1. Make changes to [Magic Modules](https://github.com/GoogleCloudPlatform/magic-modules) to add or modify resource conversion code.
+2. Generate terraform-google-conversion.
+3. Run tests locally.
+4. Make PRs for Magic Modules with your changes.
+
+Each of these is discussed in more details below.
+
+#### 1. Magic Modules
+
+Magic Modules uses a shared code base to generate terraform-google-conversion and the [google](https://github.com/hashicorp/terraform-provider-google) and [google-beta](https://github.com/hashicorp/terraform-provider-google-beta) Terraform providers.
+Most Terraform resources are represented as [yaml files which are grouped by product](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/mmv1/products).
+Each product has an `product.yaml` file (which defines the basic product information) and  Resource.yaml files (which defines any resource-specific information).
+A Resource.yaml file can specify `include_in_tgc_next_DO_NOT_USE: true` to enable converters autogeneration, or `exclude_resource: true` to skip autogeneration for both converters and the providers.
+
+Auto-generating converters code based on yaml files is strongly preferred.
+
+##### Handwritten converters
```

Comment:
Let's make the handwritten vs generated sections work like [this](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/develop/add-resource.md?plain=1)